### PR TITLE
Extend search in *_default_os fixtures for EL9 as default OS

### PR DESCRIPTION
**Pull request by Gauravtalreja1**
### Problem Statement
When EL9 is default OS for Foreman, *_default_os fixtures returns below error 
```
pytest_fixtures/component/os.py:24: in default_os
    os = entities.OperatingSystem().search(query={'search': search_string})[0].read()
E   IndexError: list index out of range
```
### Solution
Add  EL9 in search_string for *_default_os fixtures
Move default os search query to constants and delete unused session_default_os fixture (from review)

### CODE REVIEW

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6638
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_registration.py::test_positive_global_registration_end_to_end[rhel9] --external-logging
Test Result : ================= 1 passed, 15 warnings in 1104.39s (0:18:24) ==================
```

File: pytest_fixtures/component/os.py
Comment (Gauravtalreja1) [reply pytest_fixtures/component/os.py]: I wonder how tests from test_contentview_old is working even when Foreman is on EL8, and second one from api_factory helper is unused so its expected we missed it for EL8 earlier.

@jameerpathan111 Good point, about moving this search query as robottelo contant, so we can update this at single place when OS version changes.  I've made the changes, request to re-review, Thank you!

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_registration.py::test_positive_global_registration_end_to_end[rhel9]

File: pytest_fixtures/component/os.py
Comment (jyejare) [inline pytest_fixtures/component/os.py:22]: FYI, there is a [discussion/PR](https://github.com/SatelliteQE/robottelo/pull/14670#discussion_r1567443179) in progress on how we retrieve the default OS.

We are to remove the need of search string especially for searching default os.

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6639
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/ui/test_registration.py::test_positive_global_registration_end_to_end[rhel9] --external-logging
Test Result : ================= 1 passed, 15 warnings in 1160.57s (0:19:20) ==================
```

**Comment by ogajduse** [comment]
I would like to remind the audience of this PR that there is an open issue pointing to the need to add newly released versions to the list. My personal preference would be to address the issue rather than continue using the same pattern.
https://github.com/SatelliteQE/robottelo/issues/10481

**Comment by jameerpathan111** [comment]
> I would like to remind the audience of this PR that there is an open issue pointing to the need to add newly released versions to the list. My personal preference would be to address the issue rather than continue using the same pattern. #10481

@Gauravtalreja1 what you wanna do? you wanna take up this task or go with your current approach?

File: pytest_fixtures/component/os.py
Comment (jameerpathan111) [reply pytest_fixtures/component/os.py:22]: @jyejare I don't see much progress on the discussion there. Do we want to hold off on this PR/fix? Is someone already assigned to work on it?

**Review by jameerpathan111 — APPROVED** [review]

File: pytest_fixtures/component/os.py
Comment (jyejare) [reply pytest_fixtures/component/os.py:22]: @jameerpathan111 @Gauravtalreja1  can either choose to fix [discussion/PR](https://github.com/SatelliteQE/robottelo/pull/14670#discussion_r1567443179) comment in this PR or merge this as is if its urgent.

Which we would you like to go ?

**Review by pondrejk — APPROVED** [review]

File: pytest_fixtures/component/os.py
Comment (dosas) [reply pytest_fixtures/component/os.py:22]: a ping would have been nice

Comment (dosas) [reply pytest_fixtures/component/os.py:22]: > @jyejare I don't see much progress on the discussion there. Do we want to hold off on this PR/fix? Is someone already assigned to work on it?

2 weeks ago there were two approvals and no answer from @lpramuk requested changes

**Pull request merged**
